### ISSUE TITLE
What should I do? If I want permuted_axes move data.

### ISSUE THREAD

**Issue by Qinka** [rust-ndarray/ndarray#1205]
the document says:

pub fn [permuted_axes](https://docs.rs/ndarray/latest/ndarray/struct.ArrayBase.html#method.permuted_axes)<T>(self, axes: T) -> [ArrayBase](https://docs.rs/ndarray/latest/ndarray/struct.ArrayBase.html)<S, D>
where
    T: [IntoDimension](https://docs.rs/ndarray/latest/ndarray/trait.IntoDimension.html)<Dim = D>, 
Permute the axes.

This does not move any data, it just adjusts the array’s dimensions and strides.

But, what should I do, if I want permuted_axes to move data?

My shape is [1, 3, 244, 244], but strides is [150528, 1, 672, 3], but I need [150528, 50176, 224, 1].

In other words, I want to know, ``how to make c- or f-contiguous''

**Comment by jturner314** [comment]
The simplest solution is to allocate a new array of the desired layout with e.g. `Array::zeros` or `Array::uninit` and then copy the data into the new array:

```rust
use ndarray::prelude::*;

fn main() {
    let a = array![[1, 2, 3], [4, 5, 6]];
    assert_eq!(a.shape(), &[2, 3]);
    assert_eq!(a.strides(), &[3, 1]);
    assert!(a.is_standard_layout());
    
    // `b` is a Fortran-layout array, obtained by permuting the axes of a C-layout array.
    let b = a.permuted_axes([1, 0]);
    assert_eq!(b.shape(), &[3, 2]);
    assert_eq!(b.strides(), &[1, 3]);
    assert!(!b.is_standard_layout());
    
    // Create a new, C-layout array.
    let mut c = Array2::uninit(b.raw_dim());
    // Copy the data into the new array.
    b.assign_to(&mut c);
    let c = unsafe { c.assume_init() };
    assert_eq!(c.shape(), &[3, 2]);
    assert_eq!(c.strides(), &[2, 1]);
    assert!(c.is_standard_layout());
    assert_eq!(b, c);
    
    println!("done");
}
```

**Comment by Qinka** [comment]
I finally chose:
```
let a = array![[1, 2, 3], [4, 5, 6]];
let b = a.permuted_axes([1, 0]).as_standard_layout().to_owned();
```

I think it is a good idea to add some documents about `contiguous`.

**Comment by jturner314** [comment]
Oh, nice; that's a more concise solution. :) I forgot about `as_standard_layout`. In the `to_owned` docs, "contiguous" means that, when considering the linear layout of the elements in memory, there are no holes between elements of the array.
